app/services/samples/batch_file_import_service.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require 'roo'
+
+module Samples
+  # Service used to batch create samples via a file
+  class BatchFileImportService < BaseSpreadsheetImportService
+    def initialize(namespace, user = nil, blob_id = nil, params = {})
+      @sample_name_column = params[:sample_name_column]
+      @project_puid_column = params[:project_puid_column]
+      @sample_description_column = params[:sample_description_column]
+      required_headers = [@sample_name_column, @project_puid_column]
+      super(namespace, user, blob_id, required_headers, 0, params)
+    end
+
+    def execute
+      authorize! @namespace, to: :import_samples_and_metadata?
+      validate_file
+      perform_file_import
+    rescue FileImportError => e
+      @namespace.errors.add(:base, e.message)
+      {}
+    end
+
+    protected
+
+    def perform_file_import # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+      response = {}
+      parse_settings = @headers.zip(@headers).to_h
+
+      @spreadsheet.each_with_index(parse_settings) do |data, index|
+        next unless index.positive?
+
+        # TODO: handle metadata headers
+
+        sample_name = data[@sample_name_column]
+        project_puid = data[@project_puid_column]
+        description = data[@sample_description_column]
+
+        error = errors_on_sample_row(sample_name, project_puid, response, index)
+        unless error.nil?
+          response["index #{index}"] = error
+          next
+        end
+
+        project = Namespaces::ProjectNamespace.find_by(puid: project_puid)&.project
+        error = errors_with_project(project_puid, project)
+        unless error.nil?
+          response[sample_name] = error
+          next
+        end
+
+        response[sample_name] = process_sample_row(sample_name, project, description)
+      end
+      cleanup_files
+      response
+    end
+
+    private
+
+    def accessible_from_namespace?(project)
+      if @namespace.project_namespace?
+        @namespace.id == project.namespace.id
+      elsif @namespace.group_namespace?
+        authorized_scope(Project, type: :relation, as: :group_projects, scope_options: {
```

Comment:
I think we should assign this to a variable once (after a check to make sure `@namespace` is a group namespace. Right now if the namespace is a group namespace this would get called for every row in the spreadsheet

```
authorized_scope(Project, type: :relation, as: :group_projects, scope_options: {
                           group: @namespace,
                           minimum_access_level: Member::AccessLevel::UPLOADER
                         })
```

Then here we can just check to see if the authorized scope has the project id within it.

Will need to initialize the variable to nil in the initialize and then after line 46, if the project exists we can assign the above to something like `@VARIABLE_NAME ||=`